Who uses Dart (updated) (#4601)
dart-lang/site-www: src/community/who-uses-dart.md

toc: false
---

## Flutter

> The popular [Flutter toolkit][] is powered by the Dart language,
> runtimes, tooling, and libraries. Over 500,000 Flutter apps for Android,
> iOS, the web, and other targets, are all Dart-based.

Reviewer: **issue:** Active voice, per the [GSG](https://developers.google.com/style/tone).
```suggestion
The Dart language and ecosystem powers the popular [Flutter toolkit][].
Developers have created more than 500,000 Flutter apps for Android,
iOS, the web, and other targets using Dart.
```

The [Flutter showcase][] features popular Flutter apps, including:

* [BMW][]

>     > "The BMW Group is establishing itself as a leading automotive app developer

Reviewer 2: If we're keeping quotes, I think they should be ones that talk about Dart somehow.

Alternatively, we could just change this to short descriptions, and pull them up a level, e.g.:
```
* [BMW][]: A total of 300 developers use Dart to deliver a unified *My BMW app* for iOS and Android.
```

    and provider with a Flutter architecture developed 100% in-house."
* [Sonos][]: 
    > "...If setup is the “grand entryway” to the Sonos ecosystem, Flutter forms
    its elegant façade..."
* [Alibaba][]: 
    > "When the Xianyu team was challenged to improve their user experience by
    developing an app that was consistent across platforms, had beautiful graphics,
    and still loaded fast, they turned to Flutter."
* [PUBG][]: 
    > "When the PUBG MOBILE team wanted to add a community module to allow players
    from all over the world to share gameplay clips and more, they turned to Flutter."

[Flutter toolkit]: {{site.flutter}}
[Flutter showcase]: {{site.flutter}}/showcase
[BMW]: https://www.press.bmwgroup.com/global/article/detail/T0328610EN/the-my-bmw-app:-new-features-and-tech-insights-for-march-2021?language=en
[Sonos]: https://tech-blog.sonos.com/posts/renovating-setup-with-flutter/
[Alibaba]: https://flutter.dev/showcase/alibaba-group
[PUBG]: https://flutter.dev/showcase/pubg-mobile

## Google 

> Dart is used in almost every product group at Google:

Reviewer 2: Isn't this "by almost every product groups"? (product groups are people?)

> Dart is used in almost every product group at Google:

Reviewer: **issue:** Active voice.
```suggestion
Most product groups at Google use Dart:
```

> * The [Google Play Console][] lets app publishers market Android apps, and
> is written completely in Dart.
> * The [Google Ads][] team continues to grow, with hundreds of SWEs primarily
> writing Dart code.
> * The [Google Assistant][] team uses Flutter and Dart to drive the overall system
> UI of their Smart Displays.
> * Both the UI and much of the back-end code for [Google Nest Hub][] and Hub Max
> devices are written in Dart.

Reviewer: **issue:** Active voice.
```suggestion
* The Google Play team writes the [Google Play Console][] in Dart.
   It lets app publishers market Android apps.
* The [Google Ads][] team continues to grow with hundreds of
  SWEs writing Dart code.
* The [Google Assistant][] team uses Dart and Flutter to drive the
  user interface of their Smart Displays.
* The Google Nest team wrote the user interface and much of the backend code
  for [Google Nest Hub][] and Hub Max devices in Dart.
```

* The list goes on, with many more Google applications made using Dart and Flutter:
  * [Google Pay][]
  * [Google One][]
  * [Google Fibre][]
  * [Google Elections][]
  * [Google Shopping][]
  * [Cloud Search][]
  * [Family Link][]
  * [Google One][]
  * [Cloud Search][] 

> Several other Google teams are currently rewritting their projects in Flutter.
> In all, Google has thousands of developers writing millions of lines of code in Dart.

Reviewer: **issue:** Active voice. Also, spelling.
```suggestion
The number of Google teams rewriting their projects in
Dart and Flutter continues to grow.
In all, Google has thousands of developers writing millions
of lines of code in Dart.
```

[Google Play Console]: https://android-developers.googleblog.com/2020/06/introducing-new-google-play-console-beta.html
[Google Ads]: https://ads.google.com/getstarted
[Google Assistant]: https://developers.googleblog.com/2019/05/Flutter-io19.html
[Google Nest Hub]: https://store.google.com/us/product/nest_hub_2nd_gen?hl=en-US
[Google Pay]: https://pay.google.com/intl/en_in/about
[Google Fibre]: https://fiber.google.com
[Google Elections]: https://elections.google
[Google Shopping]: https://shopping.google.com
[Family Link]: https://families.google/familylink
[Google One]: https://one.google.com/about
[Cloud Search]: https://workspace.google.com/intl/en_in/products/cloud-search

> ## External

Reviewer 2: Not sure I understand how this is "External"? Should it be "Other"?

Beyond Google and Flutter, many companies use Dart for building their applications:

* [Workiva's][] next-generation client applications for their productivity suite
are built with Dart.
* [Wrike][] uses the [AngularDart][] framework to develop project management tools.
* [Blossom][] is an agile project management tool built with Dart.

> * And many more!

Reviewer: **question:** Should these not be combined with the quoted ones listed previously? If so, then move then and rewrite with active voice.

If you build production software with Dart and want to be included in this list,

> please visit [our site's GitHub] to open an issue or pull request, thanks!

Reviewer: **issue:** Strike `please` per the [GSG](https://developers.google.com/style/tone#politeness-and-use-of-please). Also, use `GitHub` as an adjective describing the repo, per the [GSG](https://developers.google.com/style/possessives#product,-feature,-and-company-names).
```suggestion
visit the [Dart.dev GitHub repo] to open an issue or pull request.
```

[Workiva's]: https://www.workiva.com/en-in
[Wrike]: https://www.wrike.com
[AngularDart]: https://angulardart.xyz/guide/setup
[Blossom]: https://blossom.co

> [our site's GitHub]: https://github.com/dart-lang/site-www

Reviewer: **issue:** To match the prior change.
```suggestion
[Dart.dev GitHub repo]: https://github.com/dart-lang/site-www
```
